zerver/views/streams.py

```diff
@@ -580,6 +582,137 @@ def you_were_just_subscribed_message(
 RETENTION_DEFAULT: str | int = "realm_default"
 
 
+@transaction.atomic(savepoint=False)
+@require_non_guest_user
+@typed_endpoint
+def add_channel(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    *,
+    channel: Annotated[Json[AddSubscriptionData], ApiParamConfig("channel")],
+    invite_only: Json[bool] = False,
+    is_web_public: Json[bool] = False,
+    is_default_stream: Json[bool] = False,
+    history_public_to_subscribers: Json[bool] | None = None,
+    message_retention_days: Json[str] | Json[int] = RETENTION_DEFAULT,
+    can_add_subscribers_group: Json[int | AnonymousSettingGroupDict] | None = None,
+    can_administer_channel_group: Json[int | AnonymousSettingGroupDict] | None = None,
+    can_send_message_group: Json[int | AnonymousSettingGroupDict] | None = None,
+    can_remove_subscribers_group: Json[int | AnonymousSettingGroupDict] | None = None,
+    subscribers: Json[list[str] | list[int]],
+    announce: Json[bool] = False,
+) -> HttpResponse:
+    realm = user_profile.realm
+    ensure_stream_does_not_exist_already(channel.name, realm)
+    color_map = {}
+    setting_groups_dict = {}
+    group_settings_map = {}
+    request_settings_dict = locals()
+    # We don't want to calculate this value if no default values are
```

Comment:
This misses the permission checks that we have for creating streams in `list_to_streams`.